src/target/linux.rs

```diff
@@ -137,6 +165,40 @@ impl LinuxTarget {
             })
             .collect())
     }
+
+    /// Returns the current snapshot view of this debugee process threads.
+    pub fn threads(
+        &self,
+    ) -> Result<Vec<Box<dyn Thread<ThreadId = i32>>>, Box<dyn std::error::Error>> {
+        let tasks: Vec<_> = Process::new(self.pid.as_raw())?
+            .tasks()?
+            .flatten()
+            .collect();
+
+        let mut result: Vec<Box<dyn Thread<ThreadId = i32>>> = vec![];
+        for task in tasks {
+            // The task page might be incomplete and we should retry
```

Comment:
This retry logic might be too overkill? I'm not even sure that we can get Incomplete for this particular case, just handling the generic error type (couldn't find a quick answer if it can happen on every page or just particular cases).

It's not really consistent with the other methods into procfs like tasks above... 

So what do you all thing I should do here? @nbaksalyar  @bjorn3

Reply:
I'm taking the retry out. If the lib user doesn't get a thread because it is incomplete, he just needs to call threads again.